x-pack/plugins/endpoint/public/embeddables/resolver/lib/tree_sequencers.ts

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export function* depthFirstPreorder<T>(root: T, children: (parent: T) => T[]): Iterable<T> {
```

Comment:
Would you mind putting comments on all exported stuff? For example you could write:

```ts
/**
 * Sequences a tree, yielding children returned by the `children` function. Sequencing is done in 'depth first preorder' fashion. See https://en.wikipedia.org/wiki/Tree_traversal#Pre-order_(NLR)
 */
export function* depthFirstPreorder<T>(root: T, children: (parent: T) => T[]): Iterable<T> {
```

Reply:
done